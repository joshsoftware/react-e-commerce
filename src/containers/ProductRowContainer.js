import React from 'react';
import RowWrapper from '../components/RowWrapper';
import ColumnWrapper from '../components/ColumnWrapper';
import PropTypes from 'prop-types';
import CardComponent from '../components/CardComponent';

const ProductRowContainer = ({ products }) => {
  let arr = [];
  for (let i = 0; i < 3; i++) {
    if (products[i] !== undefined) {
      arr.push(
        <ColumnWrapper key={products[i].id} data={<CardComponent product={products[i]} />} />
      );
    }
  }
  return <RowWrapper data={arr} />;
};

export default ProductRowContainer;

ProductRowContainer.propTypes = {
  products: PropTypes.arrayOf(PropTypes.object).isRequired
};
