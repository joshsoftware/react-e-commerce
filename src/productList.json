[
    {"id": 1, "quantity": 1, "category": "Sporting Goods", "product_price": 49.99, "stock": 10, "product_title": "Football", "description": "high quality football item", "image_url": "https://cdn.shopify.com/s/files/1/0011/7312/7180/products/jpeg-jpeg01-t-shirt-tshirt-clothing-black-cyberpunk-goth-geek-cult-23-keep-it-url-1.jpg?v=1581498404"},
    {"id": 2, "quantity": 1, "category": "Sporting Goods", "product_price": 9.99, "stock": 10, "product_title": "Baseball", "description": "high quality Baseball item", "image_url": "https://cdn.shopify.com/s/files/1/0011/7312/7180/products/jpeg-jpeg01-t-shirt-tshirt-clothing-black-cyberpunk-goth-geek-cult-23-keep-it-url-1.jpg?v=1581498404"},
    {"id": 3, "quantity": 1, "category": "Sporting Goods", "product_price": 29.99, "stock": 10, "product_title": "Basketball", "description": "high quality basketball item", "image_url": "https://cdn.shopify.com/s/files/1/0011/7312/7180/products/jpeg-jpeg01-t-shirt-tshirt-clothing-black-cyberpunk-goth-geek-cult-23-keep-it-url-1.jpg?v=1581498404"},
    {"id": 4, "quantity": 1, "category": "Electronics", "product_price": 99.99, "stock": 10, "product_title": "iPod Touch", "description": "high quality ipod touch item", "image_url": "https://cdn.shopify.com/s/files/1/0011/7312/7180/products/jpeg-jpeg01-t-shirt-tshirt-clothing-black-cyberpunk-goth-geek-cult-23-keep-it-url-1.jpg?v=1581498404"},
    {"id": 6, "quantity": 1, "category": "Electronics", "product_price": 399.99, "stock": 10, "product_title": "iPhone 5", "description": "high quality iphone item", "image_url": "https://cdn.shopify.com/s/files/1/0011/7312/7180/products/jpeg-jpeg01-t-shirt-tshirt-clothing-black-cyberpunk-goth-geek-cult-23-keep-it-url-1.jpg?v=1581498404"}
]