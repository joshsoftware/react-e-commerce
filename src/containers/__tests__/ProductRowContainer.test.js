import React from 'react';
import { render } from '@testing-library/react';
import ProductRowContainer from '../ProductRowContainer';
import productList from '../../productList.json';

describe('ProductRow Container', () => {
  it('Must return prodcuts row', () => {
    const { asFragment } = render(<ProductRowContainer products={productList.slice(0, 3)} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
