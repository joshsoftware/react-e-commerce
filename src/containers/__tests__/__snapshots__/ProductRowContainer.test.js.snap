// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductRow Container Must return prodcuts row 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <div
      class="col"
    >
      <div
        class="card"
      >
        <img
          alt="Product Image"
          class="card-img"
          src="https://cdn.shopify.com/s/files/1/0011/7312/7180/products/jpeg-jpeg01-t-shirt-tshirt-clothing-black-cyberpunk-goth-geek-cult-23-keep-it-url-1.jpg?v=1581498404"
        />
        <div
          class="card-body"
        >
          <div
            class="card-title"
          >
            Football
          </div>
          <p
            class="card-text"
          >
            $49.99
          </p>
        </div>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="card"
      >
        <img
          alt="Product Image"
          class="card-img"
          src="https://cdn.shopify.com/s/files/1/0011/7312/7180/products/jpeg-jpeg01-t-shirt-tshirt-clothing-black-cyberpunk-goth-geek-cult-23-keep-it-url-1.jpg?v=1581498404"
        />
        <div
          class="card-body"
        >
          <div
            class="card-title"
          >
            Baseball
          </div>
          <p
            class="card-text"
          >
            $9.99
          </p>
        </div>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="card"
      >
        <img
          alt="Product Image"
          class="card-img"
          src="https://cdn.shopify.com/s/files/1/0011/7312/7180/products/jpeg-jpeg01-t-shirt-tshirt-clothing-black-cyberpunk-goth-geek-cult-23-keep-it-url-1.jpg?v=1581498404"
        />
        <div
          class="card-body"
        >
          <div
            class="card-title"
          >
            Basketball
          </div>
          <p
            class="card-text"
          >
            $29.99
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
